src/super_gradients/training/datasets/detection_datasets/coco_detection_yolox.py

```diff
@@ -0,0 +1,287 @@
+                "Caching images for the first time. This might take about 20 minutes for COCO"
+            )
+            self.imgs = np.memmap(
+                cache_file,
+                shape=(len(self.ids), max_h, max_w, 3),
+                dtype=np.uint8,
+                mode="w+",
+            )
+            from tqdm import tqdm
+            from multiprocessing.pool import ThreadPool
+
+            NUM_THREADs = min(8, os.cpu_count())
+            loaded_images = ThreadPool(NUM_THREADs).imap(
+                lambda x: self.load_resized_img(x),
+                range(len(self.annotations)),
+            )
+            pbar = tqdm(enumerate(loaded_images), total=len(self.annotations))
+            for k, out in pbar:
+                self.imgs[k][: out.shape[0], : out.shape[1], :] = out.copy()
+            self.imgs.flush()
+            pbar.close()
+        else:
+            logger.warning(
+                "You are using cached imgs! Make sure your dataset is not changed!!\n"
+                "Everytime the self.input_size is changed in your exp file, you need to delete\n"
+                "the cached data and re-generate them.\n"
+            )
+
+        logger.info("Loading cached imgs...")
+        self.imgs = np.memmap(
+            cache_file,
+            shape=(len(self.ids), max_h, max_w, 3),
+            dtype=np.uint8,
+            mode="r+",
+        )
+
+    def load_resized_img(self, index):
+        """
+        Loads image at index, and resizes it to self.input_dim
+
+        :param index: index to load the image from
+        :return: resized_img
+        """
+        img = self.load_image(index)
+        r = min(self.input_dim[0] / img.shape[0], self.input_dim[1] / img.shape[1])
+        resized_img = cv2.resize(
+            img,
+            (int(img.shape[1] * r), int(img.shape[0] * r)),
+            interpolation=cv2.INTER_LINEAR,
+        ).astype(np.uint8)
+        return resized_img
+
+    def load_sample(self, index):
+        """
+        Loads sample at self.ids[index] as dictionary that holds:
+            "image": Image resized to self.input_dim
+            "target": Detection ground truth, np.array shaped (num_targets, 5), format is [class,x1,y1,x2,y2] with
+                image coordinates.
+            "target_seg": Segmentation map convex hull derived detection target.
+            "info": Original shape (height,width).
+            "id": COCO image id
+
+        :param index: Sample index
+        :return: sample as described above
+        """
+        id_ = self.ids[index]
+        res, res_seg, img_info, resized_info, _ = self.annotations[index]
+        if self.imgs is not None:
+            pad_img = self.imgs[index]
+            img = pad_img[: resized_info[0], : resized_info[1], :].copy()
+        else:
+            img = self.load_resized_img(index)
+
+        sample = {"image": img, "target": res.copy(), "target_seg": res_seg, "info": img_info, "id": np.array([id_])}
```

Comment:
will the target include the "crowd" parameter?

Reply:
Derived from the image id as far as I know.